logbook-servlet/src/main/java/org/zalando/logbook/servlet/LocalResponse.java

```diff
@@ -106,42 +114,54 @@ public PrintWriter getWriter() throws IOException {
         return body;
     }
 
-    private final class TeeServletOutputStream extends ServletOutputStream {
+    private static final class TeeServletOutputStream extends ServletOutputStream {
 
-        private final OutputStream original;
+        private final DataOutput output;
+        private final LazyStream original;
 
-        private TeeServletOutputStream() throws IOException {
-            this.original = LocalResponse.super.getOutputStream();
+        private TeeServletOutputStream(final DataOutput output,
+                final LazyStream original) {
```

Comment:
Then, to make it safe (i.e. client does not cache result of `retrieveLazily`) you may implement `class LazyStream extends OutputStream` and hide original there, so from POV of `TeeServletOutputStream` it is just another output stream.